Index: htdocs/app/code/local/Mage/SalesRule/Model/Rule/Condition/Product/Subselect.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- htdocs/app/code/local/Mage/SalesRule/Model/Rule/Condition/Product/Subselect.php	(date 1444315919000)
+++ htdocs/app/code/local/Mage/SalesRule/Model/Rule/Condition/Product/Subselect.php	(revision )
@@ -114,7 +114,7 @@
         $attr = $this->getAttribute();
         $total = 0;
         foreach ($object->getQuote()->getAllVisibleItems() as $item) {
-            if (parent::validate($item)) {
+            if (Mage_Rule_Model_Condition_Combine::validate($item)) {
                 $total += $item->getData($attr);
             }
         }
