Index: htdocs/app/code/local/Mage/Catalog/Model/Url.php
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.CharsetEP
<+>UTF-8
===================================================================
--- htdocs/app/code/local/Mage/Catalog/Model/Url.php	(date 1443013818000)
+++ htdocs/app/code/local/Mage/Catalog/Model/Url.php	(date 1443013779000)
@@ -806,7 +806,7 @@
             /**
              * Check if existing request past can be used
              */
-            if (!empty($requestPath)
+            if ($product->getUrlKey() == '' && !empty($requestPath)
                 && strpos($existingRequestPath, $requestPath) === 0
             ) {
                 $existingRequestPath = preg_replace(
